### ISSUE TITLE
ACL Parser exception raised if ACLs are not attached to any interfaces

### ISSUE THREAD

**Issue by csolernet** [CiscoTestAutomation/genielibs#174]
ACL Parser exception raised if ACLs are not attached to any interfaces

Issue while parsing: <class 'genie.libs.parser.nxos.show_acl.ShowAccessListsSummary'>

Traceback (most recent call last):
  File "src/genie/cli/commands/learn.py", line 365, in genie.cli.commands.learn.LearnCommand._retrieve_ops
  File "/home/csoler/automation/Python-NRE/lib/python3.10/site-packages/genie/libs/ops/acl/nxos/acl.py", line 239, in learn
    self.make(final_call=True)
  File "/home/csoler/automation/Python-NRE/lib/python3.10/site-packages/genie/ops/base/base.py", line 111, in make
    self.maker.make(*args, **kwargs)
  File "src/genie/ops/base/maker.py", line 315, in genie.ops.base.maker.Maker.make
  File "src/genie/ops/base/maker.py", line 427, in genie.ops.base.maker.Maker._call_parser
  File "src/genie/metaparser/_metaparser.py", line 342, in genie.metaparser._metaparser.MetaParser.parse
  File "src/genie/metaparser/_metaparser.py", line 322, in genie.metaparser._metaparser.MetaParser.parse
  File "src/genie/metaparser/util/schemaengine.py", line 419, in genie.metaparser.util.schemaengine.Schema.validate
genie.metaparser.util.exceptions.SchemaMissingKeyError: Missing keys: [['attachment_points']]

The issue is resolved as far as at least one ACL is attached to an interface, but I believe that limits the scope of the tests

Issue can be reproduced on a Cisco Nexus devices running v9.3.13 with default settings:
1.- Configure remote mgmt. to the device (SSH and credentials)
2.- Configure a simple ACL like below

config t
ip access-list MGMT-ACL
    10 remark terminal servers
    20 permit ip 10.10.10.10/32 any
    30 permit ip 10.10.10.11/32 any
    998 permit ip any any log
    999 deny ip any any l	
!

3.- Below pyats command will raise an exception when learning the ACL features:

pyats learn acl --testbed testbed.yml

4.- Once the interface is attached to an interface, the issue is resolved

config t
int eth1/1
  ip access-group MGMT-ACL in
!

**Comment by SohanTirpude** [comment]
Hello @csolernet,

It looks like the parser `ShowAccessListsSummary` expects the output which contains the interface attached. Can you please share me the raw output of `show access-lists summary` command before and after attaching an interface?

Thank you.

**Comment by csolernet** [comment]
Hi @SohanTirpude,
Thanks for looking into this issue
I do not think the attached interface should be mandatory as there might be ACLs used for other purposes (like routing policies) or not active at all

Please, find the below the output of those commands

Kind regards,
César

`n9kv02(config-if)# do sh access-list MGMT-ACL

IP access list MGMT-ACL
        10 remark terminal servers
        20 permit ip 10.10.10.10/32 any
        30 permit ip 10.10.10.11/32 any
        998 permit ip any any log
        999 deny ip any any log

n9kv02(config-if)# do sh access-list summary

IPV4 ACL MGMT-ACL
        Total ACEs Configured: 4
        Configured on interfaces:
        Active on interfaces:

n9kv02(config-if)# ip access-group MGMT-ACL in
n9kv02(config-if)# do sh access-list summary

IPV4 ACL MGMT-ACL
        Total ACEs Configured: 4
        Configured on interfaces:
                Ethernet1/1 - ingress (Router ACL)
        Active on interfaces:
`

**Comment by sowmyadn010501** [comment]
Hi, 
I got this output after making attached interface optional, Kindly check whether it's proper or not for below output
```
IPV4 ACL MGMT-ACL
Total ACEs Configured: 4
Configured on interfaces:
Active on interfaces:
```
```
{
    "acl": {
        "MGMT-ACL": {
            "total_aces_configured": 4
        }
    }
}
```

**Comment by csolernet** [comment]
Hi @sowmyadn010501 

That looks perfect to me! It seems like the object does raise any errors if no attached interfaces are present

Thanks a million for looking into this issue!

KInd regards,
César

**Comment by SohanTirpude** [comment]
Hello @csolernet, 

The PR has been merged and the fix will be available in the upcoming pyats version 24.8. Hence, I am closing this ticket for now.
Kindly try again after updating pyats to 24.8 and let me know if this same issue is seen. If you are facing any other issues, kindly open a new ticket.
